src/types/github.ts

```diff
@@ -97,7 +97,7 @@ export type RawGitTreeEntry = {
   path: string
   mode: string
   type: "tree" | "file"
-  sha: string
+  sha: string | null
```

Comment:
so sha is optional? would `sha?` work?

Reply:
No, `sha?` doesn't work because there are places where `sha` is assigned specifically to `null`. 
```
 newGitTree.push({
            ...item,
            sha: null,
          })
```
If we do `sha? string` that's equivalent to `sha: string | undefined` which cannot be assigned to `null`.

Reply:
got it